service/dap/server.go

```diff
@@ -752,22 +752,22 @@ func (s *Server) onScopesRequest(request *dap.ScopesRequest) {
 		// the amount of the returned data manageable. In fact, this is
 		// considered so expensive even with the package filter, that
 		// the default for showGlobalVariables was recently flipped to
-		// not showing. If we delay loading of the globals until the
-		// corresponding entry is expanded, generating an explicit variable
-		// request, should we consider making all globals accessible by package?
+		// not showing. If we delay loading of the globals until the corresponding
+		// scope is expanded, generating an explicit variable request,
+		// should we consider making all globals accessible with a scope per package?
 		// Or users can just rely on watch variables.
-		currPkg := "??"
-		var globals []api.Variable
 		if state.CurrentThread != nil && state.CurrentThread.File != "" {
+			currPkg := "??"
 			currDir := path.Dir(filepath.ToSlash(state.CurrentThread.File)) // Normalized across OS
 			// Retrieve package-to-compiled-directory mapping, so we can match
 			// the directory of the current file with the breakpoint to the import
 			// path of its package. The import path can then be used to filter
 			// all global package variables returned by the debugger by default
 			// down to only the ones in the current package.
 			pkgInfo := s.debugger.ListPackagesBuildInfo( /*includeFiles*/ false)
```

Comment:
As per offline discussion with @aarzilli, I am looking into replacing this file-to-package logic with parsing the package name from the current function name. 

I have access to state.CurrentThread.Function, which is of type *api.Function. I believe it is converted from *proc.Function in api.ConvertFunction(). And proc.Function has PackageName. I can expand api.Function to include package name separately from the fully qualified Name (as well as cu.isgo) and copy those over in api.ConvertFunction().

Reply:
I was thinking of just adding a CurrentPackage method to debugger.(*Debugger) actually.

Reply:
How about now?